tests/InferTypesTest.elm

```diff
@@ -1,106 +1,151 @@
 import TestHelpers exposing (located)
 
 
 typeInference : Test
 typeInference =
     let
-        runSection ( description, tests ) =
+        runSection : String -> List ( String, Canonical.Expr, Result Error.TypeError Type ) -> Test
+        runSection description tests =
             describe description
                 (List.map runTest tests)
 
+        runTest : ( String, Canonical.Expr, Result Error.TypeError Type ) -> Test
         runTest ( description, input, output ) =
             test description <|
                 \() ->
                     located input
                         |> Stage.InferTypes.inferExpr
                         |> Result.map Typed.getType
                         |> Expect.equal output
+
+        fuzzExpr : Type -> Test
+        fuzzExpr typeWanted =
+            fuzz (Fuzz.exprTyped typeWanted) (dumpType typeWanted) <|
+                \input ->
+                    Stage.InferTypes.inferExpr input
+                        |> Result.map Located.unwrap
+                        |> Result.map Tuple.second
+                        |> Expect.equal (Ok typeWanted)
+
+        fuzzExpressions : String -> List Type -> Test
+        fuzzExpressions description types =
+            types
+                |> List.map fuzzExpr
+                |> describe description
     in
     describe "Stage.InferType"
-        (List.map runSection
-            [ ( "list"
-              , [ ( "empty list"
-                  , Canonical.List []
-                  , Ok (List (Var 1))
-                  )
-                , ( "one item"
-                  , Canonical.List
-                        [ located (Canonical.Literal (Literal.Int 1)) ]
-                  , Ok (List Int)
-                  )
-                , ( "more items"
-                  , Canonical.List
-                        [ located (Canonical.Literal (Literal.Int 1))
-                        , located (Canonical.Literal (Literal.Int 2))
-                        , located (Canonical.Literal (Literal.Int 3))
-                        ]
-                  , Ok (List Int)
-                  )
-                , ( "different types"
-                  , Canonical.List
-                        [ located (Canonical.Literal (Literal.Int 1))
-                        , located (Canonical.Literal (Literal.String "2"))
-                        ]
-                  , Err (TypeMismatch Int String)
-                  )
-                , ( "more items with different types"
-                  , Canonical.List
-                        [ located (Canonical.Literal (Literal.Bool True))
-                        , located (Canonical.Literal (Literal.String "two"))
-                        , located (Canonical.Literal (Literal.Int 3))
-                        ]
-                  , Err (TypeMismatch Bool String)
-                  )
-                ]
+        [ runSection "list"
+            [ ( "empty list"
+              , Canonical.List []
+              , Ok (Type.List (Type.Var 1))
```

Comment:
We can try to drop the `Type.` qualifiers in the expected outputs.
`Ok (List (Var 1))` etc.

Reply:
We could even go with `Ok "List a"` if we call `dumpType` in the test. :thinking:

Reply:
I'd probably rather keep the `Type` type there, not call `dumpType` :) but yeah, we totally could!

Reply:
The newest version here uses `dumpType`, just to see how that looks. Pro: it reads nice. Con:  breaking `dumpType` will break these tests too.

Reply:
I don't know... I see it looks nicer, but I'd still rather have the AST there as the expected output than the String.